src/letsRoute.js

```diff
@@ -26,131 +26,153 @@ var wildcardRegexStr = "\\/([^\\/:]*)\\*";
         }
         return;
     }
 
-    var matches = getAllMatches(url,paramRegexStr);
-    if(matches.length > 0){//DYNAMIC
+    var matches = getAllMatches(url, paramRegexStr);
+    if (matches.length > 0) {//DYNAMIC
         var paramsArr = [];
-        for(var i=0; i< matches.length; i++){
+        for (var i = 0; i < matches.length; i++) {
             var name = matches[i][1];
             var pattern = matches[i][2];
 
             paramsArr.push(name);
-            if(pattern){
-                if(name === "*" && pattern !== "(.*)" ){
+            if (pattern) {
+                if (name === "*" && pattern !== "(.*)") {
                     var breakIndex = pattern.indexOf(".*");
-                    url = url.replace(matches[i][0], pattern.substr(1,breakIndex - 1) + "(.*)" );
-                }else{
-                    url = url.replace(matches[i][0],pattern);
+                    url = url.replace(matches[i][0], pattern.substr(1, breakIndex - 1) + "(.*)");
+                } else {
+                    url = url.replace(matches[i][0], pattern);
                 }
-            }else{
-                url = url.replace(matches[i][0],"([^\\/]+)");
+            } else {
+                url = url.replace(matches[i][0], "([^\\/]+)");
             }
-            if(this.ignoreTrailingSlash){
-                if(url.endsWith("/")){
+            if (this.ignoreTrailingSlash) {
+                if (url.endsWith("/")) {
                     url = url + "?";
-                }else{
+                } else {
                     url = url + "/?";
                 }
             }
+            if (this.ignoreLeadingSlash) {
+                if (url.startsWith("/") === false) {
+                    url = "/" + url;
+                }
+            }
         }
-        var regex = new RegExp("^"+url+"$");
+        var regex = new RegExp("^" + url + "$");
         this.checkIffRegistered(this.dynamicRoutes, method, url);
-        this.dynamicRoutes[method][url] = { regex: regex, fn: fn, params: params || {}, paramsArr : paramsArr};
-    }else{//STATIC
+        this.dynamicRoutes[method][url] = { regex: regex, fn: fn, params: params || {}, paramsArr: paramsArr };
+    } else {//STATIC
         this.checkIffRegistered(this.staticRoutes, method, url);
-        this.staticRoutes[method][url] = { fn : fn, params: params };
-        if(this.ignoreTrailingSlash){
-            if(url.endsWith("/")){
-                url = url.substr(0,url.length-1);
-            }else{
+        this.staticRoutes[method][url] = { fn: fn, params: params };
+        if (this.ignoreTrailingSlash) {
+            if (url.endsWith("/")) {
+                url = url.substr(0, url.length - 1);
+            } else {
                 url = url + "/";
             }
-            this.staticRoutes[method][url] = { fn : fn, params: params };
+            this.staticRoutes[method][url] = { fn: fn, params: params };
+        }
+        if (this.ignoreLeadingSlash) {
```

Comment:
This condition should come before all the conditions or before adding the route. It's purpose to modify the URL only. So if leading slash is not present the add it otherwise skip. remove rest of the code.